test/libsolidity/syntaxTests/abiEncoder/abi_encode_conversions.sol

```diff
@@ -0,0 +1,30 @@
+interface testInterface {
+    function A(address payable) external;
+    function B(function (string calldata) external) external;
+    function C(function (string memory) external) external;
+    function D(string calldata) external;
+}
+
+contract testContract {
+    function g(string calldata) external {}
+    function h(string memory) external {}
+    string s;
+
+    function main() external view {
+        abi.encodeCall(testInterface.A, (address(0)));
+        abi.encodeCall(testInterface.B, (s));
+        abi.encodeCall(testInterface.C, (s));
+        abi.encodeCall(testInterface.D, (s));
```

Comment:
This covers the `storage`/`calldata` combination. Can you also add others for full coverage, like I mentioned in https://github.com/ethereum/solidity/pull/12750/files#r823996395?
- Encoding a `calldata` value as an input for a `calldata` argument
- Encoding a `calldata` value as an input for a `memory` argument
- Encoding a `memory` value as an input for a `calldata` argument
- Encoding a `storage` value as an input for a `memory` argument